runtime/reference/vscode.md

```diff
@@ -0,0 +1,510 @@
+import { assert } from "jsr:@std/assert@1";
+
+Deno.test({
+  name: "a test case",
+  fn() {
+    let someCondition = true;
+    assert(someCondition);
+  },
+});
+```
+
+You will see a code lens just above the test:
+
+```console
+▶ Run Test
+```
+
+If you click on the code lense, the extension will start up the Deno CLI to run
+the test for you and display the output. Based on your other settings, the
+extension will try to run your test with the same settings. If you need to
+adjust the arguments provided when doing `deno test`, you can do so by setting
+the `deno.codeLens.testArgs` setting.
+
+The extension will also try to track if in the same module you destructure the
+`Deno.test` function or assign it to a variable. So you can do something like
+this and still have the code lens work:
+
+```ts
+const { test: denoTest } = Deno;
+
+denoTest({
+  name: "example test",
+  fn() {},
+});
+```
+
+If you want to disable this feature, you can do so by unsetting the **Deno >
+CodeLens: Test** setting - `deno.codeLens.test`.
+
+## Using the debugger
+
+The extension provides integration with the built-in VS Code debugger. You can
+generate a configuration by: going to `Run and Debug` panel, clicking
+`create a launch.json file` and selecting `Deno` option from the available
+debugger options.
+
+By default, the configuration will use `--inspect-wait` flag if the configured
+Deno version is greater than 1.29, or `--inspect-brk` for versions older than
+1.29. This ensures that the debugger has a chance to connect to your program and
+register all the breakpoints specified in the code.
+
+## Tasks
+
+The extension communicates directly to the language server, but for some
+development tasks, you might want to execute the CLI directly. The extension
+provides a task definition for allowing you to create tasks that execute the
+`deno` CLI from within the editor.
+
+### Deno CLI tasks
+
+The template for the Deno CLI tasks has the following interface, which can be
+configured in a `tasks.json` within your workspace:
+
+```ts
+interface DenoTaskDefinition {
+  type: "deno";
+  // This is the `deno` command to run (e.g. `run`, `test`, `cache`, etc.)
+  command: string;
+  // Additional arguments pass on the command line
+  args?: string[];
+  // The current working directory to execute the command
+  cwd?: string;
+  // Any environment variables that should be set when executing
+  env?: Record<string, string>;
+}
+```
+
+Several of the commands that are useful in the editor are configured as
+templates and can be added to your workspace by select **Tasks: Configure Task**
+in the command palette and searching for `deno` tasks.
+
+And example of what a `deno run mod.ts` would look like in a `tasks.json`:
+
+```json
+{
+  "version": "2.0.0",
+  "tasks": [
+    {
+      "type": "deno",
+      "command": "run",
+      "args": [
+        "mod.ts"
+      ],
+      "problemMatcher": [
+        "$deno"
+      ],
+      "label": "deno: run"
+    }
+  ]
+}
+```
```

Comment:
I don't understand this whole section. Is it talking about `deno task` or how to set up tasks in vscode in general for their debugging feature?